### DISCUSSION TITLE
access local images

### DISCUSSION THREAD

**Discussion by flosigime** [sqlpage/SQLPage#96] [Q&A]
Hello, 

Is it possible to view an image that is outside sqlpage's directory and not on the web ? 
It will be usefull to list pictures just by giving the local adress. 

Thanks

**Comment by lovasoa** [comment]
No, SQLPage will not serve files outside of its web root directory (which is [configurable](https://github.com/lovasoa/SQLpage/blob/main/configuration.md)).

If you need to reference images that are outside of the root directory, you can make a simple symbolic link from outside the root directory to a destination inside it.

```sh
ln -s my_images_folder my_sqlpage_root/my_images_alias
```

and than in a sql file

```sql
SELECT 
    'hero' as component,
    'Look at my image !' as title,
    '/my_images_alias/my_nice_picture.jpg' as image,
```

**Comment by flosigime** [comment]
Thanks for your answer.

Yes, i thought about symbolic links and i just discover we can use them in
windows...i try this way

Do you have examples of webroot configuration ?
Thanks you

Le sam. 7 oct. 2023, 01:42, Ophir LOJKINE ***@***.***> a
écrit :

**Reply by lovasoa** [reply]
On windows, for instance, you can have `sqlpage/sqlpage.json`:

```json
{
  "web_root": "C:\\My Folder"
}
```

**Comment by flosigime** [comment]
Hello,

i tried using web_root, but if i use \ for the path on windows, it creates a mistake and sqlpage is not working

So i write the path just like in linux, ie :

` "web_root":"C:/Users/florent/Documents/04-Publications_PAS_Recherche/00_test_sqlpage/donnee/enregistrement/sqlpage/"`

but, if i ask for an image in

`C:/Users/florent/Documents/04-Publications_PAS_Recherche/00_test_sqlpage/donnee/image`

it does not work.

Another try :
`
"web_root":"/00_test_sqlpage/donnee/enregistrement/sqlpage/"
`


but not working.

It works of course with symbolic link...

Thank you

**Reply by lovasoa** [reply]
> i tried using web_root, but if i use \ for the path on windows, it creates a mistake and sqlpage is not working

In json strings, you have to escape backslashes with other backslashes, like in the example I gave above. `"C:\\x\\y\\z"`, not `"C:\x\y\z"`. Of, course, you can also just use forward slashed, which are supported natively in windows. 

> but, if i ask for an image in
> `C:/Users/florent/Documents/04-Publications_PAS_Recherche/00_test_sqlpage/donnee/image`

I am not sure what you mean by "ask for an image in". If your web_root is in `C:\x\y\z`, then the image is in `C:\x\y\z\picture.jpg`, is accessible on `http://localhost:8080/picture.jpg` and is referenced from inside `C:\x\y\z\index.sql` by just `/picture.jpg`, not `C:\x\y\z\picture.jpg`

**Comment by flosigime** [comment]
great !
So : 
- define your web root in sqlpage.json
- put an index.sql file in the web_root folder as defined in the json.
- you can put you sqlpage app in subfolder of your web_root and use other directory to store images or other data for example...
